packages/recommend-core/src/utils/getHitsPerPage.ts

```diff
@@ -1,17 +1,17 @@
 import { GetRecommendationsInternalProps } from '../getRecommendations';
 
 type GetHitsPerPageParams<TObject> = {
-  fallbackFilters: GetRecommendationsInternalProps<TObject>['fallbackFilters'];
+  fallbackParameters: GetRecommendationsInternalProps<TObject>['fallbackParameters'];
   maxRecommendations: GetRecommendationsInternalProps<TObject>['maxRecommendations'];
   recommendationsCount: number;
 };
 
 export function getHitsPerPage<TObject>({
-  fallbackFilters,
+  fallbackParameters,
   maxRecommendations,
   recommendationsCount,
 }: GetHitsPerPageParams<TObject>) {
-  const hasFallback = fallbackFilters.length > 0;
+  const hasFallback = fallbackParameters.facetFilters!.length > 0;
```

Comment:
We don't necessarily have `fallbackParameters.facetFilters` anymore so this line can throw.

```suggestion
  const hasFallback = Array.isArray(fallbackParameters.facetFilters) && fallbackParameters.facetFilters.length > 0;
```